src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Parsing/ConnectionString/OdbcConnectionStringParser.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using NewRelic.Agent.Helpers;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+
+namespace NewRelic.Agent.Extensions.Parsing.ConnectionString
+{
+    public class OdbcConnectionStringParser : IConnectionStringParser
+    {
+        private static readonly List<string> _hostKeys = new List<string> { "server", "data source", "hostname" };
+        private static readonly List<string> _portKeys = new List<string> { "port" };
+        private static readonly List<string> _databaseNameKeys = new List<string> { "database" };
+
+        private readonly DbConnectionStringBuilder _connectionStringBuilder;
+
+        public OdbcConnectionStringParser(string connectionString)
+        {
+            _connectionStringBuilder = new DbConnectionStringBuilder { ConnectionString = connectionString };
+        }
+
+        public ConnectionInfo GetConnectionInfo(string utilizationHostName)
+        {
+            var host = ParseHost();
+            if (host != null) host = ConnectionStringParserHelper.NormalizeHostname(host, utilizationHostName);
+            var portPathOrId = ParsePortPathOrId();
+            var databaseName = ConnectionStringParserHelper.GetKeyValuePair(_connectionStringBuilder, _databaseNameKeys)?.Value;
+            var instanceName = ParseInstanceName();
+            return new ConnectionInfo(host, portPathOrId, databaseName, instanceName);
+        }
+
+        private string ParseHost()
+        {
+            var host = ConnectionStringParserHelper.GetKeyValuePair(_connectionStringBuilder, _hostKeys)?.Value;
+            if (host == null) return null;
+
+            // Example of want we would need to process: win-database.pdx.vm.datanerd.us,1433\SQLEXPRESS
+            try
```

Comment:
This try/catch block seems unnecessary. Looks like the code is already handling all the edge cases that are possible.